config/file_config.go

```diff
@@ -92,336 +111,137 @@ type PrometheusMetricsConfig struct {
-	c.SetDefault("HoneycombAPI", "https://api.honeycomb.io")
-	c.SetDefault("Logger", "logrus")
-	c.SetDefault("LoggingLevel", "debug")
-	c.SetDefault("Collector", "InMemCollector")
-	c.SetDefault("Metrics", "honeycomb")
-	c.SetDefault("SendDelay", 2*time.Second)
-	c.SetDefault("BatchTimeout", libhoney.DefaultBatchTimeout)
-	c.SetDefault("TraceTimeout", 60*time.Second)
-	c.SetDefault("MaxBatchSize", 500)
-	c.SetDefault("SendTicker", 100*time.Millisecond)
-	c.SetDefault("UpstreamBufferSize", libhoney.DefaultPendingWorkCapacity)
-	c.SetDefault("PeerBufferSize", libhoney.DefaultPendingWorkCapacity)
-	c.SetDefault("MaxAlloc", uint64(0))
-	c.SetDefault("HoneycombLogger.LoggerSamplerEnabled", false)
-	c.SetDefault("HoneycombLogger.LoggerSamplerThroughput", 5)
-	c.SetDefault("AddHostMetadataToTrace", false)
-	c.SetDefault("AddRuleReasonToTrace", false)
-	c.SetDefault("EnvironmentCacheTTL", time.Hour)
-	c.SetDefault("GRPCServerParameters.MaxConnectionIdle", 1*time.Minute)
-	c.SetDefault("GRPCServerParameters.MaxConnectionAge", time.Duration(0))
-	c.SetDefault("GRPCServerParameters.MaxConnectionAgeGrace", time.Duration(0))
-	c.SetDefault("GRPCServerParameters.Time", 10*time.Second)
-	c.SetDefault("GRPCServerParameters.Timeout", 2*time.Second)
-	c.SetDefault("AdditionalErrorFields", []string{"trace.span_id"})
-	c.SetDefault("AddSpanCountToRoot", false)
-	c.SetDefault("CacheOverrunStrategy", "resize")
-	c.SetDefault("SampleCache.Type", "legacy")
-	c.SetDefault("SampleCache.KeptSize", 10_000)
-	c.SetDefault("SampleCache.DroppedSize", 1_000_000)
-	c.SetDefault("SampleCache.SizeCheckInterval", 10*time.Second)
-	c.SetDefault("StressRelief.Mode", "never")
-	c.SetDefault("StressRelief.ActivationLevel", 75)
-	c.SetDefault("StressRelief.DeactivationLevel", 25)
-	c.SetDefault("StressRelief.StressSamplingRate", 100)
-	c.SetDefault("StressRelief.MinimumActivationDuration", 10*time.Second)
-	c.SetDefault("StressRelief.StartStressedDuration", 3*time.Second)
-	c.SetDefault("AdditionalAttributes", make(map[string]string))
-	c.SetDefault("TraceIdFieldNames", []string{"trace.trace_id", "traceId"})
-	c.SetDefault("ParentIdFieldNames", []string{"trace.parent_id", "parentId"})
-
-	c.SetConfigFile(config)
-	err := c.ReadInConfig()
-
+	Mode                      string   `default:"never" validate:"required,oneof= always never monitor"`
+	ActivationLevel           uint     `default:"90" validate:"gte=0,lte=100"`
+	DeactivationLevel         uint     `default:"75" validate:"gte=0,lte=100"`
+	StressSamplingRate        uint64   `default:"1000" validate:"gte=1"`
+	MinimumActivationDuration Duration `default:"10s"`
+	StartStressedDuration     Duration `default:"3s"`
+}
+
+// newConfig does the work of creating and loading the start of a config object
+// from the given arguments; if args is nil, it uses the command line arguments.
+// It's used by both the main init as well as the reload code.
+func newConfig(opts *CmdEnv) (*fileConfig, error) {
```

Comment:
Should this be named `newFileConfig`?  Looks like its goal is to return a fileConfig struct vs `NewConfig` which returns the Config interface.